### PR TITLE
Adds support for tasks triggering off event logs

### PR DESCRIPTION
**Pull request by TheSkorm**
This pull request adds support for setting up a task that is triggered from a Windows event log.

### CODE REVIEW

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.chef.io/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## Non-GitHub Verified Committers

There are 1 commit author(s) whose commits are authored by a non-GitHub verified email address. Chef will have to manually verify that they are authorized to contribute.

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by TheSkorm** [comment]
CLA signed under the username mwheeler.

File: providers/task.rb [hunk 1]
```diff
@@ -45,6 +45,8 @@
     options['RL'] = 'HIGHEST' if @new_resource.run_level == :highest
     options['IT'] = '' if @new_resource.interactive_enabled
     options['D'] = @new_resource.day if @new_resource.day
+    options['EC'] = "\"#{@new_resource.channel_name}\" " unless @new_resource.channel_name.nil?
+  	options['MO'] = "\"#{@new_resource.event_modifier}\" " unless @new_resource.event_modifier.nil?
```
Comment (ksubrama) [inline providers/task.rb]: ```options['MO']``` is already being computed above.  We should probably add ```:on_event``` as a new ```frequency``` attribute that affects the schedule and perform validation here to ensure that we aren't providing any other flags that conflict with ```:on_event``` and further check that EC isn't provided when frequency is anything but ```:on_event```.

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.chef.io/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## Non-GitHub Verified Committers

There are 1 commit author(s) whose commits are authored by a non-GitHub verified email address. Chef will have to manually verify that they are authorized to contribute.

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by tas50** [comment]
We've deprecated the windows_task resource in this cookbook in favor of the new windows_task resource in core Chef. I'm going to close this out at this point. If you'd like to see this same functionality in core Chef feel free to open up a PR there.

**Pull request closed without merging**